src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.Shared.xml

```diff
@@ -12,5 +12,8 @@
     <type fullname="System.LocalAppContextSwitches">
       <method signature="System.Boolean get_EnableUnsafeUTF7Encoding()" body="stub" value="false" feature="System.Text.Encoding.EnableUnsafeUTF7Encoding" featurevalue="false" />
     </type>
+    <type fullname="System.Resources.ResourceReader">
+      <method signature="System.Boolean InitializeBinaryFormatter()" body="stub" value="false" feature="System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization" featurevalue="false" />
```

Comment:
This shouldn't be on `ResourceReader`, the substitution should be on `System.LocalAppContextSwitches::get_BinaryFormatterEnabled()`. That way all other code that is conditioned on `LocalAppContextSwitches.BinaryFormatterEnabled` gets trimmed as well.

Reply:
We should not need `LocalAppContextSwitches::get_BinaryFormatterEnabled` in CoreLib at all. That's why I have suggested the substitution to be on `InitializeBinaryFormatter`

Reply:
Ah, Jan, I think I see what you mean. One sec while I send a new iteration.